Improved install guide

cgranade opened the pull request.

tcNickolas commented on articles/quantum-InstallConfig.md:
```diff
@@ -12,19 +12,16 @@ ms.topic: article
+> [!NOTE]
+> The Quantum Development Kit requires a 64-bit installation of Windows 10, macOS, or Linux.
 
+### Optional Prerequisites
```
I would suggest moving this section to after installing IDE and QDK; since they are optional, they should be lower priority than the main workflow

tcNickolas commented on articles/quantum-InstallConfig.md:
````diff
@@ -33,34 +30,54 @@ ms.topic: article
-You can download Visual Studio Code for free for your platform.
+1. Please see [Getting started with .NET](https://www.microsoft.com/net/learn/get-started) for how to download and install the .NET Core SDK.
+2. Once the .NET Core SDK is installed, run the following command in your favorite command line (e.g.: PowerShell or Bash) to download the latest templates for creating new Q# applications and libraries:
+   ```bash
````
This code snippet doesn't render right in preview

tcNickolas commented on articles/quantum-InstallConfig.md:
````diff
@@ -33,34 +30,54 @@ ms.topic: article
+   ```
 1. Go to the [Visual Studio Code website](https://code.visualstudio.com/).
 2. Pick the download for your platform and follow the installation prompts.
+    > [!NOTE]
````
This note doesn't render right in preview

tcNickolas commented:
"Install libraries for non-commercial use" instruction looks a bit weird - it's only mentioned in VS 2017 guide, and if somebody follows the instruction precisely, they can't run teleport because they have Quantum-NC.sln open.

tcNickolas commented:
VS 2017 guide mentions that you'll be prompted to install F# among missing features, but VS Code only mentions QDK and C# - does VS Code not want to install F#?

tcNickolas commented on articles/quantum-InstallConfig.md:
```diff
@@ -33,34 +30,54 @@ ms.topic: article
 
-## Installing and Validating the Q# Development Environment
+You should now have the Quantum Development Kit installed and ready to use in your own applications and libraries.
+The next section will cover how to build the samples and standard libraries provided along with the Quantum Development Kit.
```
Is this phrase redundant? It just repeats the next section tile

cgranade replied:
I think it helps the prose read better across that transition, but I can reword it, then.

tcNickolas replied:
I still don't like it... Together with the header and the first sentence of the next paragraph it's repeated three times.

opbld30 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/quantum-InstallConfig.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-InstallConfig.md) | :white_check_mark:Succeeded | |
[articles/quantum-WriteAQuantumProgram.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-WriteAQuantumProgram.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C9%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201807090031000249-274%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=uhVPkvCZcXTdbJvflfLk8w8mOeOnyrDpMcam8MSFe3c%3D&st=2018-07-09T00%3A27%3A24Z&se=2018-08-09T00%3A32%3A24Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

opbld38 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/quantum-InstallConfig.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-InstallConfig.md) | :white_check_mark:Succeeded | |
[articles/quantum-WriteAQuantumProgram.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-WriteAQuantumProgram.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C9%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201807090033562201-274%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=%2F78uPFXjGHQyzSQ8S2iEhN7ljlgPIw6CWKRhMnDMKV8%3D&st=2018-07-09T00%3A30%3A21Z&se=2018-08-09T00%3A35%3A21Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

opbld32 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/quantum-InstallConfig.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-InstallConfig.md) | :white_check_mark:Succeeded | |
[articles/quantum-WriteAQuantumProgram.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-WriteAQuantumProgram.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C9%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201807090036358519-274%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=GUpRYsoqu4RI1z0crMuffjFJRhAP%2BHn8BwkDkGXp1eE%3D&st=2018-07-09T00%3A32%3A59Z&se=2018-08-09T00%3A37%3A59Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

opbld39 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/quantum-InstallConfig.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-InstallConfig.md) | :white_check_mark:Succeeded | |
[articles/quantum-WriteAQuantumProgram.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-WriteAQuantumProgram.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C9%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201807090045012681-274%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=O%2BbPKbLscpfVIVWjW3Be5SI%2BKlSGFd3yk6y7A8zh0Q8%3D&st=2018-07-09T00%3A41%3A25Z&se=2018-08-09T00%3A46%3A25Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

cgranade commented:
@tcNickolas: Thanks for looking over it, I think I've addressed your comments now. Please let me know if you're good for me to merge this in. Thanks!

With respect to F# and VS Code, the .NET Core SDK already provides F# support. If a user wants to use F# from within VS Code, then the Ionide extensions work quite well for that. By contrast, F# compilation and editor support are coupled in VS2017, so you kind of need for that to be there.

tcNickolas commented on articles/quantum-InstallConfig.md:
```diff
@@ -126,3 +132,15 @@ In this section you will clone the quantum samples & libraries repository, and r
+- The Python interoperability feature has been developed for the [Anaconda distribution](https://www.anaconda.com/download/) of Python 3.6.
+  Please see the [README](https://github.com/Microsoft/Quantum/tree/master/Samples/PythonInterop/README.md) file provided with the Python sample for more details.
+
+Additionally, the Quantum Development Kit is provided along with libraries for non-commercial use.
```
Is QDK actually provided with these libraries, or are these libraries provided separately? I suspect there is a line to be drawn here. I'd say something like "You can find more Q# libraries for non-commercial use at ..."

cgranade replied:
Since the non-commercial phase estimation operations are documented with the rest of the QDK APIs, I think it makes sense to keep the phrasing as is for now, with an eye to revisiting as you suggest.

opbld31 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/quantum-InstallConfig.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-InstallConfig.md) | :white_check_mark:Succeeded | |
[articles/quantum-WriteAQuantumProgram.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/cgranade/improved-install-guide/articles/quantum-WriteAQuantumProgram.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C9%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201807091549293495-274%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=7vW3qdscyXmjUC4uw%2FbhANwm%2Bc6kbn07%2F9J2mklDVeY%3D&st=2018-07-09T15%3A45%3A54Z&se=2018-08-09T15%3A50%3A54Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.
